ankihub/db/utils.py

```diff
@@ -6,10 +6,10 @@ class TimedLock:
     """Thread lock with a timeout. Raises a RuntimeError if the lock could not be acquired within the timeout."""
 
     def __init__(self, timeout_seconds: float):
-        self.lock: threading.Lock = threading.Lock()
+        self.lock: threading.RLock = threading.RLock()
```

Comment:
Changed this into a re-entrant lock because `_AnkiHubDB.remove_deck()` is now calling _AnkiHubDB.remove_note_types_of_deck()`, and both of them use `_AnkiHubDB.write_lock`. Without this change there would be a deadlock in `_AnkiHubDB.remove_deck()`